### PR TITLE
JS UI test for useModal=true

### PR DESCRIPTION
**Pull request by bartzet**
# Problem
UI test for useModal=true is in Ruby

# Solution
Implement automated test in JS, remove Ruby one together with step definition and page object.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [ ] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [ ] Have tests passed locally?
- [ ] Have any new strings been translated?

### CODE REVIEW

File: test/js/pageobjects/Welcome.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ class Welcome extends Base {
   get subtitle() { return this.$('.onfido-sdk-ui-Welcome-text')}
   get footer() { return this.$('.onfido-sdk-ui-Theme-footer')}
   get primaryBtn() { return this.$('.onfido-sdk-ui-Button-button')}
+  get openModalButton() { return this.$('#button')}
+  get closeButton() { return this.$('.onfido-sdk-ui-Modal-closeButton')}
```
Comment (kopijunkie) [inline test/js/pageobjects/Welcome.js]: Minor: recommend this is renamed to `closeModalButton` for consistency & to be clear what it is/does

File: test/js/pageobjects/Welcome.js [hunk 2]
```diff
@@ -34,6 +36,14 @@ class Welcome extends Base {
   async verifyFooter() {
     this.footer.isDisplayed()
   }
+
+  async clickOnButtonToOpenModalView() {
```
Comment (kopijunkie) [inline test/js/pageobjects/Welcome.js]: for consistency with the selector's name should be named `clickOnOpenModalButton`

Comment (bartzet) [reply test/js/pageobjects/Welcome.js]: Done in: 6bba9d1

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-737-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-737-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-737-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-737-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-737-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-737-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://737-pr-onfido-sdk-ui-onfido.surge.sh

File: test/js/specs/happy.js [hunk 3]
```diff
@@ -587,4 +587,19 @@ describe('Happy Paths', options, ({driver, pageObjects}) => {
...
+    it('should be able to open, close and open again a modal view', async () => {
+      driver.get(localhostUrl + `?useModal=true`)
+      const welcomeCopy = welcome.copy()
+      welcome.clickOnOpenModalButton()
+      welcome.verifyTitle(welcomeCopy)
+      driver.sleep(500)
+      welcome.clickOnCloseModalButton()
+      driver.sleep(500)
+      welcome.clickOnOpenModalButton()
+      welcome.verifyTitle(welcomeCopy)
```
Comment (kopijunkie) [inline test/js/specs/happy.js]: I think this test is fine as it is, but maybe there should be another test for the modal to close on `Esc` key press?

Comment (bartzet) [reply test/js/specs/happy.js]: I will add it in the separate ticket.

**Pull request merged**